src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -160,45 +147,29 @@ internal ITelemetryReceiver TelemetryReceiver
         /// Unless requested otherwise by Microsoft support, this parameter should not be set by application developers as it may have adverse effect on the application.
         /// This property is also concatenated to the <c>extraQueryParameter</c> parameters of token acquisition operations.
         /// </summary>
-        public string SliceParameters { get; set; }
+        public string SliceParameters => ServiceBundle.Config.SliceParameters;
+        // TODO: VALIDATE IF WE NEED THIS VARIABLE
 
         /// <Summary>
         /// Token Cache instance for storing User tokens.
         /// </Summary>
-        internal TokenCache UserTokenCache
-        {
-            get => _userTokenCache;
-            set
-            {
-                _userTokenCache = value;
-                if (_userTokenCache != null)
-                {
-                    _userTokenCache.ClientId = ClientId;
-                    _userTokenCache.ServiceBundle = ServiceBundle;
-                }
-            }
-        }
+        internal TokenCache UserTokenCache { get; }
 
         /// <summary>
         /// Gets/sets a boolean value telling the application if the authority needs to be verified against a list of known authorities. The default
         /// value is <c>true</c>. It should currently be set to <c>false</c> for Azure AD B2C authorities as those are customer specific
         /// (a list of known B2C authorities cannot be maintained by MSAL.NET). This property can be set just after the construction of the application
         /// and before an operation acquiring a token or interacting with the STS.
         /// </summary>
-        public bool ValidateAuthority { get; set; }
-
-        /// <summary>
-        /// ExtendedLifeTimeEnabled is a Boolean that first party applications (read Office) can set to true in case when the STS has an outage,
-        /// to be more resilient.
-        /// </summary>
-        private bool ExtendedLifeTimeEnabled { get; set; }
+        public bool ValidateAuthority => ServiceBundle.Config.DefaultAuthorityInfo.ValidateAuthority;
+        // TODO: VALIDATE IF WE NEED THIS VARIABLE
```

Comment:
Need feedback on whether we actually need this to be publicly exposed on clientapplicationbase.  i'm not sure what purpose it serves for developers to read this.

Reply:
I don't know the history on this, but for b2c, we're automatically setting it to false now (as of a couple releases ago), so devs no longer need to set it for B2C authorities. B2C seemed to be the only use case for setting it to false. 

---
In reply to: [241546011](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/723#discussion_r241546011) [](ancestors = 241546011)